File: presentation/src/main/java/com/khalturin/dmitriy/presentation/di/Injector.java
package com.khalturin.dmitriy.presentation.di;

import android.app.Application;

import com.khalturin.dmitriy.presentation.di.component.AppComponent;
import com.khalturin.dmitriy.presentation.di.component.DaggerAppComponent;
import com.khalturin.dmitriy.presentation.di.module.ApplicationContextModule;
import com.khalturin.dmitriy.presentation.di.module.NavigatorModule;

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 10.04.18 1:26.
 */

public final class Injector {

//==================================================================================================
//    Class Variables
//==================================================================================================

  private static Injector sInstance = new Injector();

  private static AppComponent sAppComponent;

//==================================================================================================
//    Class Methods
//==================================================================================================

  public static Injector getInstance(){
    return sInstance;
  }

  public void buildAppComponent(Application application){
    sAppComponent = DaggerAppComponent.builder()
      .applicationContextModule(new ApplicationContextModule(application))
      .navigatorModule(new NavigatorModule())
      .build();
  }

  public AppComponent getAppComponent(){
    return sAppComponent;
  }

}

File: rssreader/src/main/kotlin/dmitriykhalturin/rssreader/di/Injector.kt
package dmitriykhalturin.rssreader.di

import android.app.Application
import android.support.v4.app.FragmentActivity
import dmitriykhalturin.rssreader.di.component.ApplicationComponent
import dmitriykhalturin.rssreader.di.component.DaggerApplicationComponent
import dmitriykhalturin.rssreader.di.component.DaggerPresenterComponent
import dmitriykhalturin.rssreader.di.module.ApplicationModule
import dmitriykhalturin.rssreader.di.module.PresenterModule
import dmitriykhalturin.rssreader.di.module.RepositoryModule
import dmitriykhalturin.rssreader.di.module.StateModule

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86></dmitry.halturin.86>@gmail.com>
 * for saferide-android on 06.08.18 15:31.
 */
class Injector {

  companion object {

    @JvmStatic
    private val sInstance = Injector()

    @JvmStatic
    private lateinit var sAppComponent: ApplicationComponent

    @JvmStatic
    fun getsInstance() = sInstance

  }

  fun buildAppComponent(application: Application) {
    sAppComponent = DaggerApplicationComponent
      .builder()
      .applicationModule(ApplicationModule(application))
      .repositoryModule(RepositoryModule())
      .stateModule(StateModule())
      .build() !!
  }

  fun getsApplicationComponent() = sAppComponent

  fun getPresenterComponent(activity: FragmentActivity) = DaggerPresenterComponent
    .builder()
    .presenterModule(PresenterModule(activity))
    .build() !!

}
